lib/irq.c

```diff
@@ -217,7 +222,12 @@ irqs_set_data_bool(vfu_ctx_t *vfu_ctx, struct vfio_irq_set *irq_set, void *data)
 
     for (i = irq_set->start, d8 = data; i < (irq_set->start + irq_set->count);
          i++, d8++) {
-        efd = vfu_ctx->irqs->efds[i];
+        if (irq_set->index == VFIO_PCI_ERR_IRQ_INDEX)
```

Comment:
Were you able to test any of these changes? I'm guessing not?

Reply:
I test irqs_set_data_eventfd in my case. It works.

Reply:
You mean, you did a unit test? Is there any way we can add it to the python component tests?

Reply:
I did not a unit test. I just used it in my emulated device implementation. We can test it by running test_device_set_irqs.py.

Reply:
Does this mean you have some other changes pending for supporting your device? Since you can't currently actually generate these types of IRQs with the current API

Reply:
Actually, my device will only use msi/x irq. But qemu will set err irq and my device will use err irq as msi/x irq mistakenly. After adding this patch, this bug will be fixed.

Reply:
@jlevon do you mean that ` vfu_irq_trigger()` would also have to take the `index`?

Reply:
https://github.com/nutanix/libvfio-user/issues/387

Yes. But if qemu client side is doing it, maybe that doesn't matter much.